src/Twig/UnitsExtension.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+namespace App\Twig;
+
+use Twig\Extension\AbstractExtension;
+use Twig\TwigFunction;
+
+class UnitsExtension extends AbstractExtension
+{
+    public function getFunctions(): array
+    {
+        return [
+            new TwigFunction('is_imperial', [$this, 'isImperial']),
+            new TwigFunction('m_or_f', [$this, 'm_or_f']),
+            new TwigFunction('kph_or_mph', [$this, 'kph_or_mph']),
+            new TwigFunction('km_or_mil', [$this, 'km_or_mil']),
+        ];
+    }
+
+    public function isImperial(string $locale): bool
```

Comment:
not sure the logic to decide if we are in imperial or metric mode should stay here in the view, but ok for now